File: dmoj/gfsspc1p5/Main.java
import java.util.*;

public class Main {
    static Scanner input = new Scanner(System.in);
    static int[] coords;

    public static void main(String[] args) {
        int N = input.nextInt(), T = input.nextInt();
        coords = new int[N];
        for (int i = 0; i < N; i++) coords[i] = input.nextInt();
        Arrays.sort(coords);
        int ans = Integer.MAX_VALUE;
        for (int i = 0; i <= N - T; i++) {
            int lo = coords[i];
            int hi = coords[i + T - 1];
            int dist = Math.min(Math.abs(lo), Math.abs(hi)) + (hi - lo);
            ans = Math.min(ans, dist);
        }
        System.out.println(ans);
    }
}

File: dmoj/gfsspc1p5/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#define INFINITY 99999999
using namespace std;

int N, T;
vector<int> coords;

int main() {
    scanf("%d%d", &N, &T);
    for (int i = 0; i < N; i++) {
        int ui;
        scanf("%d", &ui);
        coords.push_back(ui);
    }
    int ans = INFINITY;

    sort(coords.begin(), coords.end());

    for (int i = 0; i <= N - T; i++) {
        int a = coords[i];
        int b = coords[i + T - 1];
        ans = min(ans, min(abs(a), abs(b)) + (b - a));
    }

    printf("%d", ans);

    return 0;
}